pkg/secrets/gcp.go

```diff
@@ -1,8 +1,32 @@
 package secrets
 
+import (
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/kanisterio/kanister/pkg/secrets/repositoryserver"
+)
+
 const (
 	// GCPProjectID is the config map key for gcp project id data
 	GCPProjectID string = "gcp_project_id"
 	// GCPServiceKey is the config map key for gcp service key data
 	GCPServiceKey string = "gcp_service_key"
+	// GCPServerAccountJsonKey is the key for gcp service account json
+	GCPServiceAccountJsonKey string = "service-account.json"
+
+	GCPSecretType string = "secrets.kanister.io/gcp"
 )
+
+func ValidateGCPCredentials(secret *v1.Secret) error {
+	if string(secret.Type) != GCPSecretType {
+		return errors.New("Secret is not GCP secret")
```

Comment:
```suggestion
		return errors.New("Secret is not a GCP compatible secret")
```
Should we suggest what the type should be as well?